### PR TITLE
WIP init netcore support

### PR DESCRIPTION
**Pull request by alexpantyukhin**
issue https://github.com/akkadotnet/Hyperion/issues/3

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
What build command needs to be executed to build this?

**Comment by alexpantyukhin** [comment]
dotnet.exe restore
dotnet.exe build Hyperion\projects.json

File: Hyperion.Tests/ExpressionTests.cs [hunk 1]
```diff
@@ -337,7 +337,7 @@ public void CanSerializeParameterExpression()
             }
         }
 
-        [Fact]
+        [Fact(Skip= "constructor Exception(SerializationInfo info, StreamingContext context) is not implemented in .netcore1.0")]
```
Comment (Horusiath) [inline Hyperion.Tests/ExpressionTests.cs]: Maybe just add a compiler flag here?

Comment (alexpantyukhin) [inline Hyperion.Tests/ExpressionTests.cs]: Yes, i'm agree. i will fix it.

**Comment by alexpantyukhin** [comment]
@Horusiath done.

**Comment by Aaronontheweb** [comment]
@alexpantyukhin we need to lower the version of .NET Standard we support. No one is using 1.6 right now, but if we supported 1.2 then we could still support virtually all of our current users **and** be used in .NET Core applications. Could we please retarget to that?

**Comment by alexpantyukhin** [comment]
@Aaronontheweb Yes sure. I will retarget it.

**Comment by alexpantyukhin** [comment]
@Aaronontheweb  Retarget is done.

**Comment by Aaronontheweb** [comment]
@alexpantyukhin 

```
[11:16:49]	[Step 1/1] Unable to resolve 'Hyperion' for '.NETFramework,Version=v4.5.2'.
[11:16:49]	[Step 1/1] Unable to resolve 'Hyperion' for '.NETCoreApp,Version=v1.0'.
[11:16:49]	[Step 1/1] Committing restore...
[11:16:49]	[Step 1/1] Lock file has not changed. Skipping lock file write. Path: D:\work\3c488133590c9ebd\Hyperion.Tests\project.lock.json
[11:16:49]	[Step 1/1] Generating MSBuild file Hyperion.Tests.nuget.targets.
[11:16:49]	[Step 1/1] D:\work\3c488133590c9ebd\Hyperion.Tests\Hyperion.Tests.csproj
[11:16:49]	[Step 1/1] Restore failed in 993ms.
[11:16:49]	[Step 1/1] 
[11:16:49]	[Step 1/1] 
[11:16:49]	[Step 1/1] NuGet Config files used:
[11:16:49]	[Step 1/1] Errors in D:\work\3c488133590c9ebd\Hyperion.Tests\Hyperion.Tests.csproj
[11:16:49]	[Step 1/1]     Unable to resolve 'Hyperion' for '.NETFramework,Version=v4.5.2'.
[11:16:49]	[Step 1/1]     Unable to resolve 'Hyperion' for '.NETCoreApp,Version=v1.0'.
[11:16:49]	[Step 1/1]     D:\work\3c488133590c9ebd\.nuget\NuGet.Config
[11:16:49]	[Step 1/1]     C:\Windows\system32\config\systemprofile\AppData\Roaming\NuGet\NuGet.Config
[11:16:49]	[Step 1/1]     C:\ProgramData\nuget\Config\Microsoft.VisualStudio.Offline.config
[11:16:49]	[Step 1/1] 
[11:16:49]	[Step 1/1] Feeds used:
[11:16:49]	[Step 1/1]     https://api.nuget.org/v3/index.json
[11:16:49]	[Step 1/1]     C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
[11:16:49]	[Step 1/1] 
[11:16:49]	[Step 1/1] Installed:
[11:16:49]	[Step 1/1]     120 package(s) to D:\work\3c488133590c9ebd\Hyperion.Tests\Hyperion.Tests.csproj
[11:16:49]	[Step 1/1] Task failed with Package restore of .\Hyperion.sln failed
[11:16:49]	[Step 1/1] Retry.
```

**Comment by Aaronontheweb** [comment]
Looks like some remaining issues with the NuGet output targets

**Comment by alexpantyukhin** [comment]
I have added cake support.
forr build: buildcake.bat -Target BuildRelease
run tests: buildcake.bat -Target RunTests
create nuget: buildcake.bat -Target CreateNuget
publish nuget: buildcake.bat -Target PublishNuget

Currently there are still fake files and old nuspecs.

**Comment by Aaronontheweb** [comment]
@alexpantyukhin can you add a task for "all" which will run the build, tests, and nuget packaging?

**Comment by alexpantyukhin** [comment]
@Aaronontheweb Done! Also I have described how to run `PublishNuget` target in the my prev comment.

File: Hyperion/project.json [hunk 2]
```diff
@@ -18,20 +22,22 @@
...
+        },
         "Microsoft.CSharp": "4.0.1"
-      }
+      },
+      "imports": [
+        "netstandard1.2"
+      ]
     },
-    "net450": {
+    "net452": {
```
Comment (alexvaluyskiy) [inline Hyperion/project.json]: Why have you changed net45 to net452? It will be incompatible with Akka

File: Hyperion/project.json [hunk 3]
```diff
@@ -18,20 +22,22 @@
   },
 
   "frameworks": {
-    "netstandard1.6": {
-      "imports": "dnxcore50",
+    "netcoreapp1.0": {
```
Comment (alexvaluyskiy) [inline Hyperion/project.json]: We need netcoreapp ONLY for executables, not for libraries

**Comment by alexvaluyskiy** [comment]
> @Aaronontheweb  No one is using 1.6 right now, but if we supported 1.2 then we could still support virtually all of our current users and be used in .NET Core applications.
That is not true. For all our current users we could use net45 target. For all new users we could use netstandard1.6 (NetCore and Xamarin support it). For UWP apps we have to use UAP10.0 target, not netstandard.

File: Hyperion/project.json (hunk 2 above)
Comment (alexpantyukhin) [inline Hyperion/project.json]: Yes, sure! There should be net45.

File: build.cake [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    foreach(var project in projects)
+    {
+        DotNetCorePack(project.GetDirectory().FullPath, new DotNetCorePackSettings {
+            Configuration = configuration,
+            OutputDirectory = outputNuGet
+        });
+    }
+
+    // Cake - .NET 4.5
+    NuGetPack("./Hyperion/nuspec/Hyperion.nuspec", new NuGetPackSettings {
```
Comment (alexvaluyskiy) [inline build.cake]: We don't need nuspec files in NetCore projects, We could use project.json for that purposes

**Comment by alexvaluyskiy** [comment]
@alexpantyukhin Microsoft plans to release the new version of Dotnet CLI probably in the next month, So I think we should start using MSBuild projects instead of project.json.

**Comment by alexpantyukhin** [comment]
@alexvaluyskiy Yes, agree. As I know they wanted to povide automatic migration from project.json to MSBuild.

**Comment by alexvaluyskiy** [comment]
@alexpantyukhin It is not so easy. The new dotnet.exe is not compatible with the preview one. And it has a lot of bugs at the moment

**Pull request closed without merging**